### PR TITLE
`build/pkgs/nauty`: Handle `SAGE_FAT_BINARY`

### PR DESCRIPTION
**Pull request by mkoeppe**
Handling `SAGE_FAT_BINARY` as suggested by @tornaria in https://github.com/sagemath/sage/pull/36696#issuecomment-1806937921

Disabling popcnt is consistent with the handling of SAGE_FAT_BINARY in our `primesieve` install script.

Bumping the patch level so it gets recompiled when applied as a CI hotfix.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -4,8 +4,8 @@ title: SageMath
 abstract: SageMath is a free open-source mathematics software system.
 authors:
 - name: "The SageMath Developers"
-version: 10.2.rc1
+version: 10.2.rc2
```
Comment (dimpase) [inline CITATION.cff]: Don't touch this file, it's updated automatically by the release manager, from CITATION.cff.in

**Review by dimpase — CHANGES_REQUESTED** [review]

**Comment by dimpase** [comment]
ditto for [build/pkgs/sage_conf/install-requires.txt](https://github.com/sagemath/sage/pull/36707/files#diff-4e1f779d0a29296d14dfa441ec789c02f41e6cf34fae52b3b5c28c8798409330)

and other `build/pkgs/sage_*/install-requires.txt` - these changes should not be in a PR.

**Comment by mkoeppe** [comment]
I know; this is actually all coming from Volker's force pushes to develop. I'll rebase again once stabilized

File: src/doc/en/reference/references/index.rst
Comment (dimpase) [inline src/doc/en/reference/references/index.rst:1]: I think your branch has a lot of stuff from not yet released rc2, such as this file...

**Comment by dimpase** [comment]
> I know; this is actually all coming from Volker's force pushes to develop. I'll rebase again once stabilized

hmm, why does then e.g. #36699 branch look fine?

**Comment by mkoeppe** [comment]
@vbraun 10.2.rc2 is still pointing to a commit not on the develop branch
```
$ git fetch --tags -f upstream
From https://github.com/sagemath/sage
 t [tag update]              10.2.rc1   -> 10.2.rc1
$ git --no-pager log --oneline 10.2.rc0..upstream/develop --first-parent
23f671571c3 (upstream/develop) Updated SageMath version to 10.2.rc2
883e05f8ee0 (vbraun/develop) gh-36698: src/sage/combinat/root_system/coxeter_group.py: Fix typo in lazy_import
a0872bed5c9 gh-36671: build/pkgs/openblas: Stop openblas from using explicit `make -j N`; but use `make -j 1` on `ubuntu-trusty`
405e166238e gh-36669: Sage knows Hadamard matrices <1201, not 1001
bb7cf9b7c64 gh-36529: src/sage/misc/latex.py: fix view()
e349b002499 (tag: push, tag: 10.2.rc1) Updated SageMath version to 10.2.rc1
81bf4e52c66 gh-36668: CI Linux: Fix "optional", "experimental" jobs
dfde868e2f2 gh-36663: Upgrade `zeromq` to 4.3.5, `pyzmq` to 25.1.1, patch out broken tests in setup
7cad4fc0026 gh-36661: .github/workflows/doc-build.yml: Fix live doc building
781dbff5cb4 gh-36659: prompt_toolkit: Set version constraint for conda
a88fe91eae2 gh-36652: src/sage/misc/sageinspect.py: Fix pycodestyle complaint
9d2750c52bd gh-36648: CI macOS: Update
07ea7e18328 gh-36636: Deploy live doc on push to develop
88984731816 gh-36599: sage-env: identify the version of command-line tools (OS X) and set LDFLAGS accordingly
31a1ae59c46 gh-36534: CI Linux: Fix `centos-7` after #36435, remove `gentoo-python3.12` for now, use `conda-forge` with `-python3.11`
b7d34011471 gh-36533: `pkgs/sage-conf_pypi`: Repair after #36400, #36435
$ git --no-pager log --oneline --first-parent 10.2.rc1^..10.2.rc2
cd205f6e3eb (tag: 10.2.rc2) Updated SageMath version to 10.2.rc2
883e05f8ee0 (vbraun/develop) gh-36698: src/sage/combinat/root_system/coxeter_group.py: Fix typo in lazy_import
a0872bed5c9 gh-36671: build/pkgs/openblas: Stop openblas from using explicit `make -j N`; but use `make -j 1` on `ubuntu-trusty`
405e166238e gh-36669: Sage knows Hadamard matrices <1201, not 1001
bb7cf9b7c64 gh-36529: src/sage/misc/latex.py: fix view()
e349b002499 (tag: push, tag: 10.2.rc1) Updated SageMath version to 10.2.rc1
```

**Comment by mkoeppe** [comment]
Rebased onto 10.2.rc1 until there's clarity

**Review by dimpase — APPROVED** [review]
lgtm

**Comment by mkoeppe** [comment]
Thanks!

**Comment by vbraun** [comment]
For the record, the correct rc2 is:
```
$ git ls-remote github | grep 4430e4cc37c
4430e4cc37c968b71de6d2e4bf52e8825b42c719	HEAD
4430e4cc37c968b71de6d2e4bf52e8825b42c719	refs/heads/develop
4430e4cc37c968b71de6d2e4bf52e8825b42c719	refs/tags/10.2.rc2^{}
```

**Comment by mkoeppe** [comment]
Thanks, Volker, it's all good now

**Pull request merged**